dataset_generator/Fuzzer.py

```diff
@@ -90,13 +93,39 @@ def generate_print(self) -> Union[List, None]:
             self.generate_expr()
         ])
 
-    def generate_statement(self):
-        return random.choice(self.statements)()
+    def generate_func(self) -> List:
+        func_name = random.choice(func_names_pool)
+        param = self.fresh_var()
+        self.env.define_func(func_name)
+        self.env.push()
+        self.env.set_var(param)
+        body = [self.generate_statement(self.func_statements) for _ in range(random.randint(1, 3))]
```

Comment:
You're getting a lot of magic numbers here (like number of statements), I would recommend making them explicit.